projects/RabbitMQ.Client/client/impl/Connection.cs

```diff
@@ -416,9 +421,67 @@ internal void Write(ReadOnlyMemory<byte> memory)
 
         internal ValueTask WriteAsync(ReadOnlyMemory<byte> memory)
         {
+            SetNetworkTags();
             return _frameHandler.WriteAsync(memory);
         }
 
+        private void SetNetworkTags()
+        {
+            Activity? activity = Activity.Current;
```

Comment:
Wouldn't it be simpler to pass on the activity instead of reading it here? Those are all internal APIs no?

Reply:
Yes. Good catch